packages/dev/core/src/Shaders/screenSpaceReflection.fragment.fx

```diff
@@ -174,3 +196,355 @@ void main()
+                * (1.0 - max (dot(-normalize(hitCoordVS), normalize(dirVS)), 0.0)) // to fade out the reflection as the reflected direction point to the camera's position (hit behind the camera)
+                * (1.0 - currSearch) // the reflection should be sharper when near from the starting point
+                * (1.0 - clamp (abs(hitCoordVS.z / distanceFade), 0.0, 1.0)) // to fade out the reflection near the distanceFade
+                * (1.0 - clamp (deltaDepth/maxTol, 0.0, 1.0)) // since the hit point is not always precisely found, we fade out the reflected color if we aren't precise enough 
+                * clamp(1.0 - (dCoordScreen.x + dCoordScreen.y), 0.0, 1.0); // to fade out the reflection near the edge of the screen
+        }
+
+        info.visibility = clamp(info.visibility * 2.0, 0.0, 1.0); // to compensate for previous lines to have strong enough reflections
+
+        #if defined(BACKUP_TEXTURE_SKYBOX) || defined(BACKUP_TEXTURE_PROBE)
+            info.visibilityBackup = 1.0 - info.visibility; // complementary reflectivityColor
+        #endif
+        info.coords = uv;
+    }
+
+    return info;
+}
+
+void main(void)
+{
+    #ifdef SSR_SUPPORTED
+
+    // *************** Get data from samplers ***************
+
+    vec4 original = texture2D(textureSampler, vUV);
+    vec3 spec = toLinearSpace(texture2D(specularSampler, vUV).rgb);
+
+    if (dot(spec, vec3(1.0)) <= 0.0){
+        gl_FragColor = texture2D(textureSampler, vUV); // no reflectivity, no need to compute reflection
+        return;
+    }
+
+    float roughness = 1.0 - texture2D(specularSampler, vUV).a;
+
+    // Get coordinates of the direction of the reflected ray
+    // according to the pixel's position and normal.
+    vec3 unitNormal = normalize((texture2D(normalSampler, vUV)).xyz);
+    vec3 position = (view * texture2D(positionSampler, vUV)).xyz;
+
+    vec3 unitPosition = normalize(position);
+    vec3 reflected = normalize(reflect(unitPosition, unitNormal)); // incident direction = unit position in camera space
+
+    // *************** Compute reflection info  ***************
+    ReflectionInfo info;
+
+    // hash(position) represents a random vector3, jitt represents a bias to simulate roughness (light deviation)
+    vec3 jitt = mix(vec3(0.0), hash(texture2D(positionSampler, vUV).xyz), roughness) * roughnessFactor;
+
+    #ifdef RIGHT_HANDED_SCENE
+        if (position.z < -distanceFade || distanceFade == 0.0){ // no need to compute reflection, the point we are evaluating is further than the distanceFade
+            #if defined(BACKUP_TEXTURE_SKYBOX) || defined(BACKUP_TEXTURE_PROBE)
+                info.coords = vUV;
+                info.visibility = 0.0;
+                info.miss = true;
+                info.visibilityBackup = 1.0;
+            #else
+                gl_FragColor = texture2D(textureSampler, vUV);
+                return;
+            #endif
+        } else {
+            vec2 texSize = gl_FragCoord.xy / vUV;

```

Comment:
This computation will fail when one of `vUV.x` or `vUV.y` is 0, and I think you won't always have the same result from pixel to pixel because of precision problems.

You should either pass the width/height as uniforms or use `textureSize(textureSampler)` instead.